Error message improvements

DatGuyChucky opened the pull request:
## Summary
<!-- What is this pull request for? If it fixes an issue use `close #issue-number` -->
Improved error messages for CheckFailure, Forbidden, MissingRequiredArgument, and cog specific errors in verify and RFR

close #84 
## Checklist
<!-- Put an x inside [ ] to check it, like this: [x] -->

- [x] This PR fixes an issue.
- [ ] This PR adds something new (e.g. new method or parameters).
- [ ] This PR is a breaking change (e.g. methods or parameters removed/renamed)
- [ ] This PR is **not** a code change (e.g. documentation, README, ...)

<br>

- [x] Have you tested the changes? ([pytest](https://docs.pytest.org/) & [dpytest](https://dpytest.readthedocs.io/))
- [ ] Have you followed [PEP-8](https://www.python.org/dev/peps/pep-0008/) for naming and styling?  
- [ ] Has your code been properly documented with RestructuredText docstrings?
- [ ] Have you added your changes to `CHANGELOG.md` under the `[Unreleased]` heading?
- [ ] If your code added new bot commands, have you updated `documentation.json`?

<br>

- [x] All of this code is your own, or follows the author repo's licence.

DatGuyChucky committed:
is_owner Overhaul

DatGuyChucky committed:
Verify RFR error message improvement

DatGuyChucky committed:
Add files via upload

DatGuyChucky committed:
Add files via upload

DatGuyChucky committed:
Add files via upload

DatGuyChucky committed:
Revert "Add files via upload"

This reverts commit 156b9c310c9214036d99bea026e16a6f6031c49d.

DatGuyChucky committed:
Revert "Add files via upload"

This reverts commit 1f64b6a1bfe0dd4e168ab3c048f76b83e4103112.

DatGuyChucky committed:
Fixes to RFR and Verify cogs

DatGuyChucky committed:
Error type change

DatGuyChucky committed:
Change error instance to 'Exception'

Should fix current CI issues

DatGuyChucky committed:
Merge branch 'ErrorMessageImprovements' of https://github.com/KoalaBotUK/KoalaBot into ErrorMessageImprovements

JayDwee committed:
Merge branch 'master' into ErrorMessageImprovements

JayDwee commented on koalabot.py:
```diff
@@ -149,25 +150,32 @@ async def on_command_error(ctx, error: Exception):
         await ctx.send(embed=error_embed(error_type=str(type(error).__name__),
-                                         description=str(error)+"\nPlease ensure you have administrator permissions, "
-                                                                "and have enabled this extension."))
+                                         description=str(error)+"\nThe command could not be executed with the given "
+                                                                "parameter."))
```
I don't believe this change helps convey the issue very well, it seems to be a bit too abstract.

JayDwee commented on koalabot.py:
```diff
@@ -149,25 +150,32 @@ async def on_command_error(ctx, error: Exception):
+    elif isinstance(error, commands.errors.Forbidden):
+        await ctx.send(embed=error.embed(description=f"The bot lacks necessary guild permissions, re-add the bot to "
+                                                     f"this server "
+                                                     f"with the correct permissions from the README.md"))
```
README.md is not something that most users will have access to, this should instead tell the user that koala should be Administrator and placed above any other roles in the hierarchy that is assigned/ modified by koala.

JayDwee reviewed (changes requested):
unit tests need to be added to allow this to pass CodeCov check

DatGuyChucky committed:
Message slight changes

Improvements based on pull request response.

As CheckFailure is raised due to a commands.Check() error, it is unclear specifically what can cause it.

DatGuyChucky committed:
Merge branch 'ErrorMessageImprovements' of https://github.com/KoalaBotUK/KoalaBot into ErrorMessageImprovements

DatGuyChucky committed:
testing new unit test for code coverage

DatGuyChucky committed:
Fixed problematic unit test

DatGuyChucky committed:
Merge branch 'master' into ErrorMessageImprovements

DatGuyChucky committed:
Update to match master changes

DatGuyChucky committed:
Merge branch 'master' of https://github.com/KoalaBotUK/KoalaBot

DatGuyChucky committed:
Merge branch 'master' into ErrorMessageImprovements

# Conflicts:
#	koalabot.py

DatGuyChucky committed:
Merging extra stuff from master
